src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -401,4 +411,29 @@ private static String createPoolStatsInfo(String title, PoolStats poolStats) {
     }
     return title;
   }
+
+  private static Boolean bypassProxy(String account){
+    String targetHost = account + SNOWFLAKE_DOMAIN_NAME;
+    if(System.getProperty(NON_PROXY_HOSTS) != null){
+      if(isInNonProxyHosts(targetHost))
+        return true;
```

Comment:
You can just do return `return System.getProperty(NON_PROXY_HOSTS) != null && isInNonProxyHosts(targetHost)`